rust-keybroker/keybroker-client/src/lib.rs

```diff
@@ -1,17 +1,458 @@
 // Copyright 2024 Contributors to the Veraison project.
 // SPDX-License-Identifier: Apache-2.0
 
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
+use base64::engine::general_purpose::URL_SAFE_NO_PAD;
+use base64::prelude::*;
+use keybroker_common::{
+    BackgroundCheckKeyRequest, ErrorInformation, PublicWrappingKey, WrappedKeyData,
+};
+use reqwest::StatusCode;
+use rsa::{traits::PublicKeyParts, BigUint, Pkcs1v15Encrypt, RsaPrivateKey, RsaPublicKey};
+use tsm_report::{TsmReportData, TsmReportPath, TsmReportProvider};
+
+pub mod error;
+use crate::error::Error as KeybrokerError;
+use crate::error::Result;
+use crate::error::RuntimeErrorKind;
+
+/// The trait that must be implemented so a KeybrokerClient can retrieve the evidence it has
+/// to submit to the Keybroker server.
+pub trait EvidenceProvider {
+    fn get_evidence(&self, challenge: &str, verbose: bool) -> Result<Vec<u8>>;
+}
+
+/// The CCA example token.
+const CCA_EXAMPLE_TOKEN: &[u8] = &[
```

Comment:
Maybe, to make the mock more versatile (and possibly useful in debug/test situations), we could load the token from the file system rather than hardcoding it here.

Reply:
Yes, and to be honest, we have the same pattern here and there elsewhere for now (like the key name and key value and so on, or the token in the server). I agree it would be more versatile to have them in files (but there will be the question of file formats and so on). I propose to defer this improvement to some patch after this PR.

Reply:
I raised #9 to track this future enhancement.